src/Core/Hook/HookDispatcher.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Hook;
+
+use PrestaShop\PrestaShop\Adapter\Hook\HookDispatcher as HookDispatcherAdapter;
+
+/**
+ * Class HookDispatcher is responsible for dispatching hooks
+ */
+final class HookDispatcher implements HookDispatcherInterface
+{
+    /**
+     * @var HookDispatcherAdapter
+     */
+    private $hookDispatcherAdapter;
+
+    /**
+     * @param HookDispatcherAdapter $hookDispatcherAdapter
+     */
+    public function __construct(HookDispatcherAdapter $hookDispatcherAdapter)
+    {
+        $this->hookDispatcherAdapter = $hookDispatcherAdapter;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function dispatchHook(HookInterface $hook)
+    {
+        $this->hookDispatcherAdapter->dispatchForParameters(
```

Comment:
hmm, now that i think about it, maybe we actually have to return results even though this is not Rendering hook?

For example, we have a hook `paymentOptions`, it does not render anything, but rather returns `PaymentOption[]`. There are more hooks like that.

Reply:
so this should be 
```php
return $this->hookDispatcherAdapter->dispatchForParameters()...
```
instead?

Reply:
yes, as we now have `RenderedHookInterface` we could return `ExecutedHookInterface` with `getData()` which returns executed hook's data if any?

Reply:
why not, let me think about it: I'm in favor of typed hook return, but I may not want to introduce it "now".

Reply:
also, I was thinking about further refactoring:

ModuleManagementEvent should be splitted into:
* AbstractModuleEvent
* InstallModuleEvent
* UninstallModuleEvent
* ... Event

And they have to extend HookEvent instead of Event so we could call dispatchHook(AbstractModuleEvent ...) in ModuleManager: wdyt?

Reply:
hmm, i guess it makes sense to have different events for actually different events (install, uninstall & etc).